addons/pos_restaurant/static/src/app/models/restaurant_table.js

```diff
@@ -71,10 +71,13 @@ export class RestaurantTable extends Base {
         );
     }
     getOrder() {
-        return this.parent_id?.getOrder?.() || this["<-pos.order.table_id"][0];
+        return (
+            this.parent_id?.getOrder?.() ||
+            this.models["pos.order"].filter((o) => o.table_id?.id === this.id && !o.finalized)
+                .length > 0
+        );
```

Comment:
```suggestion
        return this.parent_id?.getOrder?.() || this["<-pos.order.table_id"].find((o) => !o.finalized);
```

can you try this?

Reply:
Great, I've made the changes. Please review. Thank you!